codegen/src/test/resources/software/amazon/awssdk/codegen/poet/model/serviceclientconfiguration-builder.java

```diff
@@ -176,6 +180,17 @@ public SdkClientConfiguration buildSdkClientConfiguration() {
                 internalBuilder.option(SdkClientOption.ENDPOINT, endpointOverride);
                 internalBuilder.option(SdkClientOption.ENDPOINT_OVERRIDDEN, true);
             }
+            if (credentialsProvider != null
+                    && !credentialsProvider.equals(internalBuilder.option(AwsClientOption.CREDENTIALS_IDENTITY_PROVIDER))) {
+                internalBuilder.option(AwsClientOption.CREDENTIALS_IDENTITY_PROVIDER, credentialsProvider);
+                IdentityProviders identityProviders = internalBuilder.option(SdkClientOption.IDENTITY_PROVIDERS);
+                if (identityProviders == null) {
+                    identityProviders = IdentityProviders.builder().putIdentityProvider(credentialsProvider).build();
```

Comment:
When will this be null?

Reply:
This will be `null` the first time that we build the client and run the plugins, after that it will be updated by the client itself, but not before.